code/game/objects/structures/shower.dm

```diff
@@ -252,3 +274,28 @@
 	layer = FLY_LAYER
 	anchored = TRUE
 	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+
+/obj/item/showerparts
+	name = "shower parts"
+	desc = "All the things you need to construct a shower, comes with in-built H2O atom assembler to compensate for Nanotrasen's lack of plumbing installation."
+	icon = 'icons/obj/watercloset.dmi'
+	icon_state = "showerbuild1"
+	var/created_shower_type = /obj/machinery/shower
```

Comment:
can this build off rack parts? really that functionality of `attack_self2build` should be typed similarly.

Reply:
I'm not sure how rack parts do it, I'll be sure to look into it.